frame/support/src/weights.rs

```diff
@@ -280,6 +280,20 @@ impl PostDispatchInfo {
 	}
 }
 
+/// Extract the actual weight from a dispatch result if any or fall back to the default weight.
+pub fn extract_actual_weight(result: &DispatchResultWithPostInfo, info: &DispatchInfo) -> Weight {
+	let post_info = match result {
+		Ok(post_info) => &post_info.actual_weight,
```

Comment:
```suggestion
		Ok(post_info) => post_info.actual_weight,
```

referencing seems a bit strange here..

Reply:
It was because I thought this saves one temporary copy. Not sure though. However I replaced it with @bkchr suggestion.